### PR TITLE
MOSIP-25219 / MOSIP-25755: Renamed project and moved to repo

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@RestController
+@RequestMapping("/")
+public class MockIdentityController {
+
+	@Autowired
+	private MockIdentityService mockIdentityService;
+
+	@PostMapping(value = "mock-identity", consumes = { MediaType.APPLICATION_JSON_VALUE }, produces = {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java:30]: Lets rename this to just identity in the URL.
Lets also change the controller and service names to IdentityController, IdentityService, IdentityRepository etc..

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseWrapper<MockAuthDataResponse> createMockIdentity
+	(@Valid @RequestBody RequestWrapper<MockAuthData> requestWrapper) throws MockIdentityException {
+
+		ResponseWrapper response = new ResponseWrapper<MockAuthDataResponse>();
+		MockAuthDataResponse mockAuthDataResponse = new MockAuthDataResponse();
+		mockIdentityService.saveIdentity(requestWrapper.getRequest());
+		mockAuthDataResponse.setStatus("mock auth data created successfully");
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java:38]: this is mock id data and not auth data

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/Values.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.mosip.mock.identity.system.dto;
+
+import lombok.Data;
+
+@Data
+public class Values {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/Values.java:6]: Lets rename this to LanguageValue

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		response.setResponse(mockAuthDataResponse);
+		response.setResponseTime(MockIdentityUtil.getUTCDateTime());
+		return response;
+	}
+	
+	@GetMapping(value = "mock-identity/{virtual-id}")
+	public ResponseWrapper<MockAuthData> getMockIdentity(@PathVariable(value = "virtual-id") String virtualId)
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/controller/MockIdentityController.java:45]: Lets rename virtualId to individualId everywhere in the code

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/MockAuthData.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import io.mosip.mock.identity.system.util.ErrorConstants;
+import lombok.Data;
+
+@Data
+public class MockAuthData {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/MockAuthData.java:14]: Lets name this to IdentityData

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/BiometricProperties.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package io.mosip.mock.identity.system.dto;
+
+import javax.validation.constraints.NotBlank;
+
+import io.mosip.mock.identity.system.util.ErrorConstants;
+import lombok.Data;
+
+@Data
+public class BiometricProperties {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/BiometricProperties.java:9]: Lets rename this to BiometricData

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/MockAuthDataResponse.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package io.mosip.mock.identity.system.dto;
+
+import lombok.Data;
+@Data
+public class MockAuthDataResponse {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/dto/MockAuthDataResponse.java:5]: Lets rename this to IdentityStatus

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/service/MockIdentityService.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package io.mosip.mock.identity.system.service;
+
+import io.mosip.mock.identity.system.dto.MockAuthData;
+import io.mosip.mock.identity.system.exception.MockIdentityException;
+
+public interface MockIdentityService {
+
+	public void saveIdentity(MockAuthData mockAuthDataRequest) throws MockIdentityException;
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/service/MockIdentityService.java:8]: Lets rename this to addIdentity

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/service/impl/MockIdentityServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	@Override
+	public void saveIdentity(MockAuthData mockAuthDataRequest) throws MockIdentityException {
+		MockIdentity mockIdentity = new MockIdentity();
+		try {
+			mockIdentity.setIdentityJson(objectmapper.writeValueAsString(mockAuthDataRequest));
+		} catch (JsonProcessingException e) {
+			throw new MockIdentityException(ErrorConstants.JSON_PROCESSING_ERROR);
+		}
+		mockIdentity.setIndividualId(mockAuthDataRequest.getVirtualId());
+		mockIdentityRepository.save(mockIdentity);
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/service/impl/MockIdentityServiceImpl.java:36]: In this POST endpoint, we should not allow updates, only unique individualId creates should work. If there is duplicate, we have to return with respective error in response.

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/util/MockIdentityUtil.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+import static io.mosip.mock.identity.system.util.Constants.UTC_DATETIME_PATTERN;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MockIdentityUtil {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/util/MockIdentityUtil.java:12]: Lets rename this to HelperUtil

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/util/MockIdentityUtil.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+	/**
+	 * Output format : 2022-12-01T03:22:46.720Z
+	 * 
+	 * @return Formatted datetime
+	 */
+	public static String getUTCDateTime() {
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/util/MockIdentityUtil.java:19]: rename this to getCurrentUTCDateTime

File: mock-identity-system/src/main/java/io/mosip/mock/identity/system/validator/RequestTimeValidator.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class RequestTimeValidator implements ConstraintValidator<RequestTime, String> {
+
+	@Value("${mosip.mock.authdata.reqtime.maxlimit:-2}")
+    private int maxMinutes;
+
+	@Value("${mosip..mock.authdata.reqtime.minlimit:2}")
+    private int minMinutes;
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/java/io/mosip/mock/identity/system/validator/RequestTimeValidator.java:23]: Can we take the above 2 configs as a single config called "esignet.mock.identity-system.allowed-request-time-variation-seconds" and do both + and - case.

File: mock-identity-system/src/main/resources/bootstrap.properties [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+server.servlet.context-path=/v1/mock-identity-system
+management.security.enable=false
+management.endpoint.health.show-details=always
+management.endpoints.web.exposure.include=info,health,refresh,mappings
+management.endpoint.metrics.enabled=true
+management.endpoint.prometheus.enabled=true
+management.metrics.export.prometheus.enabled=true
+registration.processor.queue.username=admin
+registration.processor.queue.password=admin
+registration.processor.queue.url=localhost
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/resources/bootstrap.properties:20]: why do we need the above 3 properties ?

File: mock-identity-system/src/main/resources/dbscripts.sql [hunk 14]
```diff
@@ -0,0 +1,5 @@
+create schema "mockidentitysystem";
+
+create table "mockidentitysystem"."mock_identity" ("individual_id" varchar(36) not null, "identity_json" varchar(2048) not null);
+
+alter table "mockidentitysystem"."mock_identity" add constraint "pk_mock_id_code" primary key ("individual_id");
```
Comment (vishwa-vyom) [inline mock-identity-system/src/main/resources/dbscripts.sql:5]: db scripts are not suppose to be outside ?

File: mock-identity-system/src/test/java/io/mosip/mock/identity/system/MockIdentitySystemApplicationTest.java [hunk 15]
```diff
@@ -0,0 +1,8 @@
+package io.mosip.mock.identity.system;
+
+/**
+ * Unit test for simple App.
+ */
+public class MockIdentitySystemApplicationTest 
+{
```
Comment (vishwa-vyom) [inline mock-identity-system/src/test/java/io/mosip/mock/identity/system/MockIdentitySystemApplicationTest.java:7]: Can we also add a few integration test scenarios which will test end to end scenarios

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: db_scripts/mosip_mockidentitysystem/drop_role.sql [hunk 16]
```diff
@@ -0,0 +1 @@
+drop role if exists idpuser;
```
Comment (ase-101) [inline db_scripts/mosip_mockidentitysystem/drop_role.sql:1]: Rename this to mockidentitysystemuser

File: db_scripts/mosip_mockidentitysystem/grants.sql [hunk 17]
```diff
@@ -0,0 +1,17 @@
+\c mosip_mockidentitysystem
+
+GRANT CONNECT
+   ON DATABASE mosip_mockidentitysystem
+   TO idpuser;
```
Comment (ase-101) [inline db_scripts/mosip_mockidentitysystem/grants.sql:5]: Rename this to mockidsystemuser

File: db_scripts/mosip_mockidentitysystem/grants.sql [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+
+GRANT USAGE
+   ON SCHEMA mockidentitysystem
+   TO idpuser;
```
Comment (ase-101) [inline db_scripts/mosip_mockidentitysystem/grants.sql:9]: Rename this to mockidsystemuser

File: db_scripts/mosip_mockidentitysystem/grants.sql [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+
+GRANT SELECT,INSERT,UPDATE,DELETE,TRUNCATE,REFERENCES
+   ON ALL TABLES IN SCHEMA mockidentitysystem
+   TO idpuser;
```
Comment (ase-101) [inline db_scripts/mosip_mockidentitysystem/grants.sql:13]: Rename this to mockidsystemuser

File: db_scripts/mosip_mockidentitysystem/grants.sql [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+
+ALTER DEFAULT PRIVILEGES IN SCHEMA mockidentitysystem
+	GRANT SELECT,INSERT,UPDATE,DELETE,REFERENCES ON TABLES TO idpuser;
```
Comment (ase-101) [inline db_scripts/mosip_mockidentitysystem/grants.sql:16]: Rename this to mockidsystemuser

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/MockIdentitySystemApplication.java [hunk 21]
```diff
@@ -0,0 +1,12 @@
+package io.mosip.esignet.mock.identitysystem;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication(scanBasePackages = { "io.mosip.mock.identity.system" })
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/MockIdentitySystemApplication.java:6]: seems like the scanbase package name is incorrect @HimajaDhanyamraju2 
it should be io.mosi.esignet.mock

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/IdentityData.java [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+import io.mosip.esignet.mock.identitysystem.util.ErrorConstants;
+import lombok.Data;
+
+@Data
+public class IdentityData {
+	
+	@NotBlank(message = ErrorConstants.INVALID_INDIVIDUAL_ID)
+	String individualId;
+
+	@NotBlank(message = ErrorConstants.INVALID_INDIVIDUAL_ID)
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/dto/IdentityData.java:19]: Pls, correct the error constant.

**Comment by vishwa-vyom** [comment]
Merging the PR based on Anusha's request.

**Pull request merged**